python/src/trezorlib/cli/ethereum.py

```diff
@@ -164,34 +167,168 @@ def _format_access_list(
     ]
 
 
+def _get_ethereum_definitions(
+    definitions_zip: Optional[pathlib.Path] = None,
+    network_def_file: Optional[BinaryIO] = None,
+    token_def_file: Optional[BinaryIO] = None,
+    download_definitions: bool = False,
+    chain_id: Optional[int] = None,
+    slip44_hardened: Optional[int] = None,
+    token_address: Optional[str] = None,
+) -> ethereum.messages.EthereumDefinitions:
+    count_of_options_used = sum(
```

Comment:
This is nice, but... we might consider creating a (data)class `EthDefinitionGetter`, take all these 4 arguments, internally do all the validation checks and then serve as a nice attribute-oriented way of telling how to get the definitions.

It would remove the need to passing so many (4) arguments so many times between functions, if they would be wrapped in one object and exposing the attributes.

The validation would also be performed earlier in the process.

Quick mock:

```python
@dataclass
class EthDefinitionGetter:
    # could have some fields as not required of course
    definitions_zip: Optional[pathlib.Path]
    network_def_file: Optional[BinaryIO]
    token_def_file: Optional[BinaryIO]
    download_definitions: bool

    def __post_init__(self):
        count_of_options_used = sum(
            bool(o)
            for o in (
                self.definitions_zip,
                (self.network_def_file or self.token_def_file),
                self.download_definitions,
            )
        )
        if count_of_options_used > 1:
            raise click.ClickException(
                "More than one mutually exclusive option for definitions was used. See --help for more info."
           )

def _get_ethereum_definitions(
    definition_getter: EthDefinitionGetter,
    chain_id: Optional[int] = None,
    slip44_hardened: Optional[int] = None,
    token_address: Optional[str] = None,
):
    if definition_getter.definitions_zip is not None:
        ...
    ...

def sign_message(
    client: "TrezorClient",
    address: str,
    message: str,
    definitions_zip: pathlib.Path,
    network_def: BinaryIO,
    download_definitions: bool,
) -> Dict[str, str]:
    """Sign message with Ethereum address."""
    address_n = tools.parse_path(address)
    defs = _get_ethereum_definitions(
        EthDefinitionGetter(
            definitions_zip=definitions_zip,
            network_def_file=network_def,
            download_definitions=download_definitions,
        ),
        slip44_hardened=address_n[1],
    )
```